tmt/utils/__init__.py

```diff
@@ -5522,7 +5522,8 @@ def dataclass_normalize_field(
     value = raw_value
 
     if dataclasses.is_dataclass(container):
-        _, _, _, _, metadata = container_field(type(container), keyname)
+        tc = type(container) if not isinstance(container, type) else container
+        _, _, _, _, metadata = container_field(tc, keyname)
```

Comment:
`tc` seems to have only purpose in life, to make the line shorter. Maybe some indentation could achieve the same, something like the following:

```suggestion
        _, _, _, _, metadata = container_field(
          type(container) if not isinstance(container, type) else container, keyname)
```

Reply:
And readability, but yeah, sure.